src/ImageContainers/DT1ImageContainer.cpp

```diff
@@ -0,0 +1,374 @@
+                return false;
+        return true;
+    }
+
+    Header::Header(LittleEndianStreamReader& fileStream)
+    {
+        static const int HEADER_SIZE = 276;
+        uint8_t zeros[260];
+        fileStream.read(version1);
+        fileStream.read(version2);
+        fileStream.read(zeros, sizeof(zeros));
+        fileStream.read(numTiles);
+        fileStream.read(tileOffset);
+        assert(allZeros(zeros, sizeof(zeros)));
+        assert(tileOffset == HEADER_SIZE);
+    }
+
+    int Tile::SUBTILE_INDEX[SUBTILE_SIZE][SUBTILE_SIZE] = {};
+    int Tile::SUBTILE_OFFSET[NUM_SUBTILES][2] = {};
+
+    /**
+     * zeros3 is not all zeros all of the time. This might correlate with unknown. I need to figure
+     * out what these variables mean, I'm fairly certain zeros3 is 3 ints, or at least the second
+     * 4 bytes are used somewhere.
+     *
+     * data\global\tiles\expansion\Town\shrine.dt1
+     * data\global\tiles\expansion\Town\trees.dt1
+     *
+     * unknown in above is same F6 62 FF 00  (16737014)
+     * but zeros3 is different:
+     * 00 00 00 00 00 00 00 00 00 00 00 00
+     * 00 00 00 00 6C B0 08 0B 00 00 00 00
+     *
+     * data\global\tiles\expansion\Town\tent.dt1
+     * F3 65 FF 00  (16737779)
+     * 00 00 00 00 4C 00 37 0B 00 00 00 00
+     */
+    Tile::Tile(LittleEndianStreamReader& fileStream)
+    {
+        // Note: static init not thread safe..
+        // TODO: Init these with C++ 17 constexpr if possible.
+        static bool staticInitDone = false;
+        if (!staticInitDone)
+        {
+            for (int y = 0, i = 0; y < SUBTILE_SIZE; y++)
+                for (int x = 0; x < SUBTILE_SIZE; x++, i++)
+                    SUBTILE_INDEX[x][y] = i;
+
+            for (int y = 0, i = 0; y < SUBTILE_SIZE; y++) {
+                int px = (WIDTH / 2) - (SUBTILE_WIDTH / 2) - (y * (SUBTILE_WIDTH / 2));
+                int py = HEIGHT - SUBTILE_HEIGHT - (y * (SUBTILE_HEIGHT / 2));
+                for (int x = 0; x < SUBTILE_SIZE; x++, i++)
+                {
+                    SUBTILE_OFFSET[i][0] = px;
+                    SUBTILE_OFFSET[i][1] = py;
+                    px += SUBTILE_WIDTH / 2;
+                    py -= SUBTILE_HEIGHT / 2;
+                }
+            }
+
+            staticInitDone = true;
+        }
+
+        uint8_t zeros1[4];
+        uint8_t zeros2[7];
+        uint8_t zeros3[12];
+        fileStream.read(direction);
+        fileStream.read(roofHeight);
+        fileStream.read(soundIndex);
+        fileStream.read(animated);
+        fileStream.read(height);
+        fileStream.read(width);
+        fileStream.read(zeros1, sizeof(zeros1));
+        orientation = Orientation(fileStream.read<int32_t>());
+        fileStream.read(mainIndex);
+        fileStream.read(subIndex);
+        fileStream.read(rarity);
+        fileStream.read(unknown, sizeof(unknown));
+        fileStream.read(flags, sizeof(flags));
+        fileStream.read(zeros2, sizeof(zeros2));
```

Comment:
Those are only used at runtime by the game afair, and one is about flags but only used during compression by D2CMP.